gprofiler/main.py

```diff
@@ -153,6 +154,7 @@ def __init__(
             profile_spawned_processes,
             bool(user_args.get("insert_dso_name")),
             profiling_mode,
+            pids_to_profile,
```

Comment:
When methods have lots of arguments, using kwargs is nicer.

Can you change the c'tor of `ProfilerState` to be kwargs-only? (i.e `__init__(self, *, ...)`) and use kwargs here? This is more readable.